pipeline/beam_tables.py

```diff
@@ -425,6 +464,70 @@ def _write_to_bigquery(self, scan_type: str,
         write_disposition=write_mode,
         additional_bq_parameters=_get_partition_params()))
 
+  def _write_to_gcs(  # pylint: disable=no-self-use
+      self, scan_type: str, rows: beam.pvalue.PCollection[BigqueryRow],
+      gcs_folder: str) -> None:
+    """Write out rows to GCS folder with hive-partioned format.
+
+    Rows are written to .json.gz files organized by source and country:
+    '{gcs_folder}/source={source}/country={country}/results.json.gz'
+
+    Args:
+      scan_type: one of 'echo', 'discard', 'http', 'https',
+        or 'satellite'
+      rows: PCollection[BigqueryRow] of data to write.
+      gcs_folder: GCS folder gs://bucket/folder/... like
+        'gs://firehook-test/scans/echo' to write output files to.
+
+    Raises:
+      Exception: if any arguments are invalid.
+    """
+
+    def _get_destination(record: str) -> str:
+      """Returns the hive-format dest folder for a measurement record str."""
+      record_dict = json.loads(record)
+      return f'source={record_dict["source"]}/country={record_dict["country"]}/results'
```

Comment:
@fortuna I'm not convinced the hive format filepaths are a good idea unless we have a strong sense of who would want to use it. It adds extra confusion for people who are just trying to read the files and parse them some other way. We can discuss in the meeting.